packages/ember-runtime/lib/system/core_object.js

```diff
@@ -53,15 +53,14 @@ function makeCtor() {
   // method a lot faster. This is glue code so we want it to be as fast as
   // possible.
 
-  let wasApplied = false;
   let initProperties, initFactory;
 
   class Class {
     constructor() {
       let self = this;
 
-      if (!wasApplied) {
-        Class.proto(); // prepare prototype...
+      if (!meta(this).parent.wasApplied) {
```

Comment:
This is a bit of a hack, I found that in some cases (particularly `ObjectProxy`) the following would be true:

```
let m1 = meta(this);
this.constructor.proto();
let m2 = meta(this);

m1 !== m2
```

For some reason, `proto` can confuse meta this way (something to do with the way `WeakMap` links between objects and values?). On the otherside, if we were to do:

```
meta(Object.getPrototypeFor(this)).wasApplied
```

The meta for a native class does not have `proto` set (it was created outside of `extend`, no chance to add it) so the check in the getter/setter for `wasApplied` fails.

I think ideally we only load meta once, and just use the `parent` value.